extensions/amp-pan-zoom/1.0/component-ts.tsx

```diff
@@ -0,0 +1,267 @@
+import {
+  GestureHandlers,
+  UserGestureConfig,
+  UserHandlers,
+  useGesture,
+} from '@use-gesture/react';
+import type {ComponentChildren, Ref} from 'preact';
+
+import {scale as cssScale, px, translate} from '#core/dom/style';
+
+import * as Preact from '#preact';
+import {useImperativeHandle, useRef} from '#preact';
+import {forwardRef} from '#preact/compat';
+import {ContainWrapper} from '#preact/component';
+import {useResizeObserver} from '#preact/hooks/useResizeObserver';
+
+import {useStyles} from './component.jss';
+import {usePanZoomState} from './hooks/usePanZoomState';
+
+const DOUBLE_TAP_TIME = 500; // Maximum time between tap starts
+
+const gestureConfig: Omit<UserGestureConfig, 'target'> = {
+  drag: {
+    filterTaps: true,
+    tapsThreshold: 10,
+  },
+};
+
+export type BentoPanZoomProps = {
+  children?: ComponentChildren;
+  controls?: boolean;
+  initialScale?: number | string;
+  initialX?: number | string;
+  initialY?: number | string;
+  maxScale?: number | string;
+};
+
+export type BentoPanZoomApi = {
+  transform(scale: number, x: number, y: number): void;
+};
+
+function getElementPosition(
+  clientX: number,
+  clientY: number,
+  element: HTMLElement
+) {
+  const elBounds = element./* REVIEW */ getBoundingClientRect();
+  return {
+    anchorX: clientX - elBounds.x,
+    anchorY: clientY - elBounds.y,
+  };
+}
+
+function classNames(...args: Array<string | false | null | 0>) {
```

Comment:
We have a `obj-str` plugin that optimizes this kind of thing.

Reply:
Updated